4/script.js

```diff
@@ -0,0 +1,164 @@
+const url = 'https://opentdb.com/api.php?amount=10';
+const answerBox = document.getElementById('answer_box');
+const subtitleBox = document.getElementById('subtitle_box');
+
+
+/*
+クイズデータ操作クラス
+*/
+class Quiz {
+  constructor(quizData) {
+    this.quizData = quizData;
+    this.quizNum = quizData.length - 1;
+    this.quizIndex = 0;
+    this.quizCorrect = 0;
+  }
+
+  //クイズインデックスを取得
+  getQuizIndex() {
+    return this.quizIndex;
+  }
+
+  //クイズインデックスを設定
+  setQuizIndex() {
+    return ++this.quizIndex;
+  }
+
+  //クイズカテゴリを取得
+  getQuizCategory() {
+    return this.quizData[this.quizIndex].category;
+  }
+
+  //クイズ難易度を取得
+  getQuizLevel() {
+    return this.quizData[this.quizIndex].difficulty;
+  }
+
+  //クイズ本文を取得
+  getQuizQuestion() {
+    return this.quizData[this.quizIndex].question;
+  }
+
+  //クイズ回答選択肢を取得
+  getQuizAnswer() {
+    const array = this.quizData[this.quizIndex].incorrect_answers;
+    return array.concat(this.quizData[this.quizIndex].correct_answer);
+  }
+
+  //クイズの正答を判定
+  checkAnser(answer) {
+    if (this.quizData[this.quizIndex].correct_answer === answer) this.quizCorrect++;
+  }
+
+  //クイズの正解数を取得
+  getQuizCorrect() {
+    return this.quizCorrect;
+  }
+}
+
+/*
+「開始」ボタン押下時の動作
+*/
+document.getElementById('start').addEventListener('click', function () {
+
+  document.getElementById('title').textContent = '取得中';
+  document.getElementById('quiz').textContent = '少々お待ちください';
+  document.getElementById('start').remove();
+
+  fetch(url)
+    .then((response) => {
+      return response.json();
+    })
+    .then(data => {
+      quizData = data.results;
+      const quiz = new Quiz(quizData);
+      makeQuiz(quiz);
+    })
+    .catch(error => {
+      console.log("失敗しました");
+    });
+});
+
+/*
+クイズ画面の生成関数
+*/
+const makeQuiz = function (quiz) {
```

Comment:
これもアロー関数でいいかなと思いました。

```suggestion
const makeQuiz = (quiz) => {
```